### ISSUE TITLE
e1000e module cannot compile with kernel-4.9.29

### ISSUE THREAD

**Issue by proteansec** [NixOS/nixpkgs#26210]
## Issue description

The e1000e module is unable to compile with the new kernel 4.9.29 as can be seen below:

> *** Downloading ‘https://cache.nixos.org/nar/1j41hlf7aarhjyafc482biwhdfhxkhk82vlwhb5rq26jlniln4m6.nar.xz’ (signed by ‘cache.nixos.org-1’) to ‘/nix/store/am9bckrgqam4mmnx5fhsm2xs3r9j9n8a-zsh-5.3.1’...
> make flags: SHELL=/nix/store/fi3mbd2ml4pbgzyasrlnp0wyy6qi48fh-bash-4.4-p5/bin/bash BUILD_KERNEL=4.9.29
> make -C /nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source O=/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/build SUBDIRS=/tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src modules
> make[1]: Entering directory '/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source'
> make[2]: Entering directory '/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/build'
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> trying http://www.mplayerhq.hu/MPlayer/releases/codecs//essential-amd64-20071007.tar.bz2
>   0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0  CC [M]  /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.o
> In file included from /nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source/include/linux/kernel.h:13:0,
>                  from /nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source/include/linux/list.h:8,
>                  from /nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source/include/linux/module.h:9,
>                  from /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.c:24:
> /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.c: In function 'e1000e_dump':
> /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.c:250:25: error: 'struct net_device' has no member named 'trans_start'
>     netdev->state, netdev->trans_start, netdev->last_rx);
>                          ^
> /nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source/include/linux/printk.h:284:34: note: in definition of macro 'pr_info'
>   printk(KERN_INFO pr_fmt(fmt), ##__VA_ARGS__)
>                                   ^
> /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.c: In function 'e1000_xmit_frame':
> /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.c:6554:8: error: 'struct net_device' has no member named 'trans_start'
>   netdev->trans_start = jiffies;
>         ^
> make[3]: *** [/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source/scripts/Makefile.build:294: /tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src/netdev.o] Error 1
> make[2]: *** [/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source/Makefile:1490: _module_/tmp/nix-build-e1000e-3.3.1.drv-0/e1000e-3.3.1/src] Error 2
> make[2]: Leaving directory '/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/build'
> make[1]: *** [Makefile:150: sub-make] Error 2
> make[1]: Leaving directory '/nix/store/w8fyxl153q32xlwaw9dqf5x5pv33m21w-linux-4.9.29-dev/lib/modules/4.9.29/source'
> make: *** [Makefile:249: default] Error 2
> builder for ‘/nix/store/l7smps6p71jq89d8ldh3fx5j6l10qyvf-e1000e-3.3.1.drv’ failed with exit code 2
> building path(s) ‘/nix/store/k464sfwgybn2yhw2asmd2xg08fls84na-etc-ssh_config’
> cannot build derivation ‘/nix/store/lqyh4dhqg8cdavliirbik5jksjnxsd12-kernel-modules.drv’: 1 dependencies couldn't be built
> killing process 9555
> killing process 9558
> killing process 9557
> killing process 9556
> killing process 9554
> killing process 9553
> building path(s) ‘/nix/store/qpaf0j21aabgddpyf6v1kf7rpwm1wyhv-keymap’
> building path(s) ‘/nix/store/cajfw3f6id958yj7sdw1dwabkdspba1c-libvirtd.conf’
> building path(s) ‘/nix/store/9fxad9szwr0fdjci5077ypc934aqgzkn-locale.conf’
> building path(s) ‘/nix/store/3pcryrz0z9rag4a93jdifmd0d824bvcc-login.defs’
> building path(s) ‘/nix/store/r8j54ijqhxpnbd6mrjl6ilqignrlj7j4-mdadm.conf’
> building path(s) ‘/nix/store/awvznm79na7g32agqilcy6xwhp4kvrpc-nix.conf’
> building path(s) ‘/nix/store/s8m9bayn4hy7qhzlq39qmgli1kk01bhi-nixos.conf’
> cannot build derivation ‘/nix/store/r6a77w1j4991asrqxcwwqrwprbv8s8af-nixos-system-user-17.03.1203.58e227052d.drv’: 1 dependencies couldn't be built
> error: build of ‘/nix/store/r6a77w1j4991asrqxcwwqrwprbv8s8af-nixos-system-user-17.03.1203.58e227052d.drv’ failed

### Steps to reproduce

Upgrade to the 17.03 NixOS release and rebuild the system.

```
# nix-channel --add https://nixos.org/channels/nixos-17.03 nixos
# nix-channel --update nixpkgs
# nixos-rebuild switch --upgrade
```

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)

> \# nixos-version 
> 16.09.1943.25f4906da6 (Flounder)

* Nix version: (run `nix-env --version`)

> \# nix-env --version
> nix-env (Nix) 1.11.8

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

> \# nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
> "17.03.1203.58e227052d"

* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)

> \# grep build-use-sandbox /etc/nix/nix.conf
> build-use-sandbox = true

**Comment by proteansec** [comment]
Solved by updating the e1000e module in hardware-configuration.nix. The relevant part of the configuration is shown below. The old version of the e1000e is not compatible with the new kernel, while the new version compiles without issues.

```
{ config, lib, pkgs, ... }:

with lib;

let
  e1000e = config.boot.kernelPackages.e1000e.overrideDerivation (attrs: {
    name = "e1000e-3.3.1";
    src = pkgs.fetchurl {
      url = "mirror://sourceforge/e1000/e1000e-3.3.5.3.tar.gz";
      sha256 = "1ajz3vdnf1y307k585w95r6jlh4ah8d74bq36gdkjl1z5hgiqi9q";
    };
  });
in {
  boot.extraModulePackages = [ e1000e ];
}
```

**Comment by pSub** [comment]
This was fixed in 163db1bbf9d24a23cae5ca828ad8c2a14b85449e in the context of 23253.
